### ISSUE TITLE
Enhancement: Add Number Entry Support at Start

### ISSUE THREAD

**Issue by drjeffries** [bitfocus/companion-satellite#138]
No idea if this is possible, but in a headless Pi install, it would be cool to be able to enter the IP address using streamdeck buttons. I'm envisioning a 0-9 and enter button to enter an ip address into the system. 

That said, now that I'm writing this, you could probably do this reverse from a post command in Companion. 

Just would be interesting to be able to connect to companion without having to pull out a laptop. 

I'd say low priority if even possible.

**Comment by Julusian** [comment]
> That said, now that I'm writing this, you could probably do this reverse from a post command in Companion.

Yes, this is something I am thinking about doing, as well as adding mdns so that companion can discover satellite if wanted

Building an ip address input is unlikely to be something I will look into, it gets a bit complex and messy for first setup. And considering that satellite supports 10+ different surface types, some of which don't have displays, or don't have enough buttons to do 0-9 input, this will likely need to be tested or implemented slightly differently for each surface type.

**Comment by sparks-alec** [comment]
Working on this! Here's progress:
![IMG_6410](https://github.com/user-attachments/assets/3de07356-da0a-4286-88d0-f0d80d51bee7)

Unfortunately I cannot get the code to commit until it lints and I'm really struggling to get ESLint and the TSConfigs to like each other. Modern JS development with its complex toolchains is very frustrating to work on.

**Comment by Julusian** [comment]
> Unfortunately I cannot get the code to commit until it lints and I'm really struggling to get ESLint and the TSConfigs to like each other. Modern JS development with its complex toolchains is very frustrating to work on.

Honestly the answer for now is probably to skip the hooks when committing.
I don't remember if I fixed the eslint issue you mention, or if I have been ignoring it by skipping the hooks
